conn.go

```diff
@@ -76,9 +76,13 @@ func (r *FilteredConn) ReadFrom(b []byte) (n int, addr net.Addr, err error) {
 	case <-timeout:
 		return 0, nil, errTimeout
 	case pkt := <-r.recvBuffer:
-		copy(b[:pkt.n], pkt.buf)
+		n := pkt.n
+		if cap(b) < n {
```

Comment:
I don’t think you can copy past the slice len, nor are expected to, so s/cap/len/?

Reply:
https://play.golang.org/p/nuyKA9ztEUe

Reply:
right, that won't work, as it will have no effect on the slice the caller has.